com.unity.netcode.gameobjects/Components/NetworkAnimator.cs

```diff
@@ -1,79 +1,169 @@
                 }
 
-                IntParameters[key] = value;
-                return true;
+                return setOrUpdatedValue;
             }
 
             public bool SetBool(int key, bool value)
             {
-                if (BoolParameters.TryGetValue(key, out var existingValue) && existingValue == value)
+                bool setOrUpdatedValue = false;
+
+                int existingKvIndex = Array.FindIndex(BoolParamArray, pair => pair.Key == key);
+
+                if (existingKvIndex == -1)
                 {
-                    return false;
+                    for (int i = 0; i < BoolParamArray.Length; i++)
+                    {
+                        var kv = BoolParamArray[i];
+
+                        if (kv.Key == k_InvalidKey)
+                        {
+                            BoolParamArray[i] = new KeyValuePair<int, bool>(key, value);
+                            setOrUpdatedValue = true;
+                            break;
+                        }
+                    }
+                }
+                else if (BoolParamArray[existingKvIndex].Value != value)
+                {
+                    BoolParamArray[existingKvIndex] = new KeyValuePair<int, bool>(key, value);
+                    setOrUpdatedValue = true;
                 }
 
-                BoolParameters[key] = value;
-                return true;
+                return setOrUpdatedValue;
             }
 
             public bool SetFloat(int key, float value)
             {
-                if (FloatParameters.TryGetValue(key, out var existingValue) &&
-                    Mathf.Abs(existingValue - value) < Mathf.Epsilon)
+                bool setOrUpdatedValue = false;
+
+                int existingKvIndex = Array.FindIndex(FloatParamArray, pair => pair.Key == key);
+
+                if (existingKvIndex == -1)
+                {
+                    for (int i = 0; i < FloatParamArray.Length; i++)
+                    {
+                        var kv = FloatParamArray[i];
+
+                        if (kv.Key == k_InvalidKey)
+                        {
+                            FloatParamArray[i] = new KeyValuePair<int, float>(key, value);
+                            setOrUpdatedValue = true;
+                            break;
+                        }
+                    }
+                }
+                else if (Math.Abs(FloatParamArray[existingKvIndex].Value - value) > Mathf.Epsilon)
+                {
+                    FloatParamArray[existingKvIndex] = new KeyValuePair<int, float>(key, value);
+                    setOrUpdatedValue = true;
+                }
+
+                return setOrUpdatedValue;
+            }
+
+            public bool SetTrigger(int key)
+            {
+                if (TriggerParameters.Contains(key))
                 {
                     return false;
                 }
 
-                FloatParameters[key] = value;
+                TriggerParameters.Add(key);
```

Comment:
I am assuming order doesn't count; if not my Set-ification would be invalid

Reply:
Yeah, order doesn't matter.